### DISCUSSION TITLE
Multiple Channel Products in Single Store Front

### DISCUSSION THREAD

**Discussion by chowadfrepo** [saleor/saleor#11283] [Q&A]
Hi! Is it possible to display all products in a single storefront that is created in Multiple Channels?
Scenario: I have multiple products in the _**mobile-channel**_   and in _**desktop-channel**_. Now I want to display all products from both channels products in the Single Store Font.

**Comment by CossackDex** [comment]
Hi @chowadfrepo,
With the `products` query, you can return products from all channels if you Authorise your request with, for example, the app with MANAGE_PRODUCTS permission, but you can't ask for channels specific details like pricing. If you are not Authorising a request or don't have the necessary permissions, you need to specify the channel.

**Reply by chowadfrepo** [reply]
Hi @CossackDex Thanks for the reply, In that case, I need to set up the authorization and also it's required to refresh the token whenever authorization expired. Which will have an impact on the performance.

**Comment by carlosa54** [comment]
@chowadfrepo currently there's not a products filter like `channel__in=["mobile-channel", "desktop-channel"]`. What you can do is to have two separate queries, one for each channel and then render the response however you want.

E.g. You can try the following query in https://demo.saleor.io/graphql/
```graphql
{
  mobileChannel: products(first: 5, channel: "default-channel") {
    edges {
      node {
        id
        name
        description
      }
    }
  }
  desktopChannel: products(first: 5, channel: "channel-pln") {
    edges {
      node {
        id
        name
        description
      }
    }
  }
}
```

Response Example:

<img width="787" alt="image" src="https://user-images.githubusercontent.com/4250593/206584026-8649118a-4c2f-4117-ab05-857efaed9488.png">

**Comment by timuric** [comment]
@chowadfrepo seems like you are not using channels as intended. Each channel requires separate checkout and has different shipping prices, and also channels have different currencies and prices.

**Reply by chowadfrepo** [reply]
@timuric yes you are right but each channel has a different seller. In that case, I want to display all sellers products into single page.
